archeryutils/handicaps/handicap_functions.py

```diff
@@ -12,754 +12,314 @@
-    # pylint: disable=too-many-arguments
-    """
-    Calculate the handicap of a given score on a given round using root-finding.
+
+def handicap_scheme(
+    handicap_sys: Union[str, HandicapScheme], **kwargs: float
+) -> HandicapScheme:
+    r"""
+    Create a HandicapScheme subclass for a requested handicap scheme.
 
     Parameters
     ----------
-    score : int or float
-        score achieved on the round
-    rnd : rounds.Round
-        a rounds.Round class object that was shot
-    hc_sys : str
-        identifier for the handicap system
-    hc_dat : handicaps.handicap_equations.HcParams
-        dataclass containing parameters for handicap equations
-    arw_d : float or None, default=None
-        arrow diameter in [metres] default = None
-    int_prec : bool, default=False
-        display results as integers? default = False, with decimal to 2dp accuracy from
-        rootfinder
+    handicap_sys : Union[str, HandicapScheme]
+        String identifying the handicap scheme class to be generated, or an existing
+        HandicapScheme
+    \\**kwargs : float
+        Various kwargs defining custom values of parameters as appropriate for hc_sys
 
     Returns
     -------
-    handicap: int or float
-        Handicap. Has type int if int_prec is True, and otherwise has type false.
+    HandicapScheme
+        a subclass of HandicapScheme as appropriate for the inputs
 
     Raises
     ------
     ValueError
-        If an invalid score for the given round is provided.
+        If a handicap scheme is requested for which there is no defined class.
+
+    See Also
+    --------
+    handicap_scheme.HandicapScheme : The HandicapScheme class
+    handicap_scheme_agb.HandicapAGB :
+        The AGB HandicapScheme subclass and associated \\**kwargs
+    handicap_scheme_agb.HandicapAGBold :
+        The AGBold HandicapScheme subclass and associated \\**kwargs
+    handicap_scheme_aa.HandicapAA :
+        The AA HandicapScheme subclass and associated \\**kwargs
+    handicap_scheme_aa.HandicapAA2 :
+        The AA2 HandicapScheme subclass and associated \\**kwargs
 
     Examples
     --------
-    Angular deviation at a distance of 25m, using the AGB handicap system at a
-    handicap of 10 can be calculated with:
+    To generate a HandicapScheme class using the Archery GB (AGB) handicap scheme:
 
-    >>> import archeryutils as au
-    >>> hc_params = au.handicap_equations.HcParams()
-    >>> wa_outdoor = au.load_rounds.WA_outdoor
-    >>> au.handicap_functions.handicap_from_score(999, wa_outdoor.wa1440_90, "AGB", hc_params)
-    43.999964586102706
-    >>> au.handicap_functions.handicap_from_score(
-    ...     999,
-    ...     wa_outdoor.wa1440_90,
-    ...     "AGB", hc_params,
-    ...     int_prec=True
-    ... )
-    44.0
+    >>> from archeryutils import handicaps as hc
+    >>> agb_scheme = hc.handicap_scheme("AGB")
+    >>> agb_scheme
+    <archeryutils.handicaps.handicap_equations.HandicapAGB object at 0x31415ee00>
```

Comment:
This looks a bit ugly showing up in the docstring, I think it shows that the Schemes need a nicer repr.
Something like `<HandicapScheme("AGB")>` `<HandicapScheme name="AGB">`, formatted to taste?

Reply:
Should be dealt with in https://github.com/jatkinson1000/archeryutils/pull/78/commits/42c67412e1b0c60a71b222b9deea8e73a3ebd704
Toyed with displaying more info, but user can always get that with more probing, so decided to adhere to KISS and use `HandicapScheme('AGB')`.

Reply:
OK, hopefully resolved with https://github.com/jatkinson1000/archeryutils/pull/78/commits/116947e5c439e8135f1cda4af17a5733d42d0a59 to address comments on the above commits.
Also added `__repr__` and `__str__` for `HandicapTable` in https://github.com/jatkinson1000/archeryutils/pull/78/commits/05b23c2cacb97be5af0ec44302e61884c33b9b9b